src/Core/Query/Helper.php

```diff
@@ -462,8 +461,24 @@ public function filterControlCharacters(string $data): string
     }
 
     /**
-     * Render placeholders in a querystring.
+     * Escape text for use as parsed character data content in an XML element.
+     *
+     * This escapes characters that can't appear as character data using their
+     * corresponding entity references. Per the definition of XML, "&" and "<"
+     * MUST be escaped when used in character data, ">" MUST be escaped in the
+     * string "]]>" and MAY be otherwise, so we escape it to be safe.
      *
+     * @param string $data
+     *
+     * @return string
+     */
+    public function escapePCDATAContent(string $data): string
```

Comment:
Escaping as `CDATA` is the other option. But the this function doesn't do it.

Reply:
Using `CDATA` sections isn't without pitfalls.

If the content of an element contains `]]>`, it has to be split up in multiple `CDATA` sections. Something like this will work:
```php
$value = '<![CDATA[' . str_replace(']]>', ']]]]><![CDATA[>', $value) . ']]>';
```

XML documents can potentially contain charachters that aren't available in the defined encoding by representing them as numeric references (`&#nnn;`). These characters cannot be represented in a `CDATA` section because it can only contain literal character data. Imported files might contain such references, so it's better to play it safe.

Transcoding during transport can also cause issues. I have no idea if this comes into play with Solr.

Reply:
In this case we should simply not name the function like this.

Reply:
Oh, I see the confusion now. I named it for [PCDATA](https://en.wikipedia.org/wiki/PCDATA), which does look a lot like [CDATA](https://en.wikipedia.org/wiki/CDATA).

Reply:
I've renamed the helper function to `escapeXMLCharacterData`. That should be less confusing.